tests/shared/drive-constraints.spec.js

```diff
@@ -92,6 +92,182 @@ describe('Shared: DriveConstraints', function() {
 
   });
 
+  describe('.isSourceDrive()', function() {
+
+    it('should return false if no image', function() {
+      const result = constraints.isSourceDrive({
+        device: '/dev/disk2',
+        name: 'USB Drive',
+        size: 999999999,
+        protected: true,
+        system: false
+      }, undefined);
+
+      m.chai.expect(result).to.be.false;
+    });
+
+    it('should return false if no drive', function() {
+      const result = constraints.isSourceDrive(undefined, {
+        path: '/Volumes/Untitled/image.img'
+      });
+
+      m.chai.expect(result).to.be.false;
+    });
+
+    it('should return false if there are no mount points', function() {
+      const result = constraints.isSourceDrive({
+        device: '/dev/disk2',
+        name: 'USB Drive',
+        size: 999999999,
+        protected: true,
+        system: false
+      }, {
+        path: '/Volumes/Untitled/image.img'
+      });
+
+      m.chai.expect(result).to.be.false;
+    });
+
+    describe('given Windows paths', function() {
+
+      it('should return true if the image lives directly inside a mount point of the drive', function() {
+        const result = constraints.isSourceDrive({
+          mountpoints: [
+            {
+              path: 'E:'
+            },
+            {
+              path: 'F:'
+            }
+          ]
+        }, {
+          path: 'E:\\image.img'
+        });
+
+        m.chai.expect(result).to.be.true;
+      });
+
+      it('should return true if the image lives inside a mount point of the drive', function() {
+        const result = constraints.isSourceDrive({
+          mountpoints: [
+            {
+              path: 'E:'
+            },
+            {
+              path: 'F:'
+            }
+          ]
+        }, {
+          path: 'E:\\foo\\bar\\image.img'
+        });
+
+        m.chai.expect(result).to.be.true;
+      });
+
+      it('should return false if the image does not live inside a mount point of the drive', function() {
+        const result = constraints.isSourceDrive({
+          mountpoints: [
+            {
+              path: 'E:'
+            },
+            {
+              path: 'F:'
+            }
+          ]
+        }, {
+          path: 'G:\\image.img'
+        });
+
+        m.chai.expect(result).to.be.false;
+      });
+
+      it('should return false if the image is in a mount point that is a substring of the image mount point', function() {
+        const result = constraints.isSourceDrive({
+          mountpoints: [
+            {
+              path: 'E:\\fo'
+            }
+          ]
+        }, {
+          path: 'E:\\foo/image.img'
+        });
+
+        m.chai.expect(result).to.be.false;
+      });
+
+    });
+
+    describe('given UNIX paths', function() {
+
+      it('should return true if the image lives directly inside a mount point of the drive', function() {
```

Comment:
Probably also worth adding a unit-test for a mountpoint being `/` and the image then being at `/image.img` ?